From 1a9d25163b6809d3342be5382d45d9489cf25e96 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 7 Nov 2023 12:12:08 +0100
Subject: [PATCH 0062/1695] plat-stm32mp1: pwr: remove test on
 CFG_DRIVERS_REGULATOR

Remove tests on CFG_DRIVERS_REGULATOR value has the config switch
is always enabled on stm32mp1 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b4a60eeb7a37be69b57232a192ad86de8a9c0036 plat-stm32mp1: pwr: remove test on CFG_DRIVERS_REGULATOR]
Change-Id: I307553a2aa50d1b114e6856c82beb87d585f7c47
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from 4a93553c32477af71590f6bfb14a21bf38249dfd]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c
index 38617a920..547cccdcd 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr.c
@@ -271,14 +271,10 @@ static TEE_Result stm32mp1_pwr_regu_probe(const void *fdt, int node,
 			panic();
 		}
 
-		if (IS_ENABLED(CFG_DRIVERS_REGULATOR)) {
-			res = regulator_dt_register(fdt, subnode, node,
-						    dt_desc + n);
-			if (res) {
-				EMSG("Can't register %s: %#"PRIx32, node_name,
-				     res);
-				panic();
-			}
+		res = regulator_dt_register(fdt, subnode, node, dt_desc + n);
+		if (res) {
+			EMSG("Can't register %s: %#"PRIx32, node_name, res);
+			panic();
 		}
 	}
 
-- 
2.47.2

